src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/EventHub/SetAzureEventHubRegenerateKeys.cs

```diff
@@ -0,0 +1,70 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Management.EventHub.Models;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.EventHub.Commands.Namespace
+{
+    [Cmdlet(VerbsCommon.Set, EventHubRegenerateKeysVerb), OutputType(typeof(ResourceListKeys))]
+    public class SetAzureEventHubRegenerateKeys : AzureEventHubsCmdletBase
```

Comment:
So, this cmdlet is regenerating the keys, right?   The verb should be new, and you can call it New-AzureRmEventHubKey.

ALso, it needs to implement ShouldProcess